core/src/test/resources/json-ld-api-tests/README.md

```diff
@@ -0,0 +1,106 @@
+
+# Design
+
+Tests driven from a top-level [manifest](manifest.jsonld) and are defined into [compact](compact-manifest.jsonld), [expand](expand-manifest.jsonld), [flatten](flatten-manifest.jsonld), [html](html-manifest.jsonld), [remote-doc](remote-doc-manifest.jsonld), [fromRdf](fromRdf-manifest.jsonld), and [toRdf](toRdf-manifest.jsonld) sections:
+
+* [compact](compact-manifest.jsonld) tests have _input_, _expected_ and _context_ documents.
+  The _expected_ results can be compared using [JSON-LD object comparison](#json-ld-object-comparison) with the processor output. Additionally, if the `ordered` option is not set, result should be expanded and compared with the expanded _expected_ document also using [JSON-LD object comparison](#json-ld-object-comparison).
+
+  For *NegativeEvaluationTests*, the result is a string associated with the expected error code.
+* [expand](expand-manifest.jsonld) tests have _input_ and _expected_ documents.
+  The _expected_ results can be compared using [JSON-LD object comparison](#json-ld-object-comparison) with the processor output.
+
+  Expansion tests may have a `expandContext` option, which is treated
+  as an IRI relative to the manifest.
+
+  For *NegativeEvaluationTests*, the result is a string associated with the expected error code.
+* [html](html-manifest.jsonld) tests have _input_ and _expected_ documents and an optional _context_ document.
+  The _expected_ results can be compared using [JSON-LD object comparison](#json-ld-object-comparison) with the processor output
+  after potentially remapping blank node identifiers (see below).
+  Additionally, if the result is compacted and the `ordered` option is not set, result should be expanded and compared with the expanded _expected_ document also using [JSON-LD object comparison](#json-ld-object-comparison).
+
+  For *NegativeEvaluationTests*, the result is a string associated with the expected error code.
+* [flatten](flatten-manifest.jsonld) tests have _input_ and _expected_ documents and an optional _context_ document.
+  The _expected_ results can be compared using [JSON-LD object comparison](#json-ld-object-comparison) with the processor output
+  after potentially remapping blank node identifiers (see below).
+  Additionally, if the result is compacted and the `ordered` option is not set, result should be expanded and compared with the expanded _expected_ document also using [JSON-LD object comparison](#json-ld-object-comparison).
+
+  For *NegativeEvaluationTests*, the result is a string associated with the expected error code.
+* [remote-doc](remote-doc-manifest.jsonld) tests have _input_ and _expected_ documents.
+  The _expected_ results can be compared using [JSON-LD object comparison](#json-ld-object-comparison) with the processor output.
+
+  For *NegativeEvaluationTests*, the result is a string associated with the expected error code.
+
```

Comment:
Hm, this sentence appears multiple times. I also can't see another outcome for NegativeEvaluationTests so defining this once should be enough.